source/common/http/filter_manager.cc

```diff
@@ -123,6 +124,11 @@ bool ActiveStreamFilterBase::commonHandleAfterHeadersCallback(FilterHeadersStatu
     // but continue filter iteration so we actually write the headers/run the cleanup code.
     headers_only = true;
     ENVOY_STREAM_LOG(debug, "converting to headers only", parent_);
+  } else if (status == FilterHeadersStatus::ContinueAndDontEndStream) {
```

Comment:
This whole else if could be replaced with a `switch`. Is there a particular reason why a `switch` isn't used?

Reply:
Probably because the states have been added incrementally over time. Feel free to change to a switch if you like.